### PR TITLE
feat: add STableCellPills component

### LINKED ISSUE #177: [Table] Error prop type on Cell component because of external type
We're getting this error.

```
[Vue warn]: Invalid prop: type check failed for prop "cell". Expected Null | Function, got Object  
  at <STableCell name="status" class-name=undefined cell= Object  ... > 
  at <STableItem key="status" name="status" class-name=undefined  ... > 
```

I think this is because we made changes in #174 and now `cell` props is defined as `TableCell | ((value: any, record: any) => TableCell)` which becomes `null | Function` because of current script setup limitation 😢 

I guess we need to redeclare TableCell type inside `STableCell` component.

### LINKED ISSUE #178: [Table] Add STableCellPills component
We need a STableCellPills component that accepts `string[]` as `value`, and show them as pills.
This is because the `STableCellPill` which we already have can't show multiple pills.

Design image is below. ("Topic type" column)
image

The usage is like below.

```ts
const table = useTable({
  ...
  columns: {
    tags: {
      label: 'Tags',
      cell: {
        type: 'pills',
        pills(items, record) {
          return items.map((item) => {
            return {
              label: item,
              color: item === 'info' ? 'info' : 'mute'
            }
          })
        }
      },
  },
  records: [
    { ... , tags: ['Info', 'News'], ... },
    { ... , tags: ['Info'], ... },
  ],
  ...
})
  ...
})
```

~Or, might be simpler to allow `STableCellPill` to accept also `string[]`?~

### PR DESCRIPTION
**Pull request by brc-dd**
fixes #178
fixes #177

### CODE REVIEW

File: stories/components/STable.story.vue [hunk 1]
```diff
@@ -85,35 +100,67 @@ const dropdownCreatedAt = createDropdown([
...
+    createdAt: '2022-09-08',
+    tags: ['Info']
+  }
 ]
 
 const filteredData = computed(() => {
   return data
     .filter((i) => filterBy(i.status, dropdownStatusSelected.value))
     .filter((i) => filterBy(i.type, dropdownTypeSelected.value))
+    .filter((i) => i.tags.some((tag) => filterBy(tag, dropdownTagsSelected.value)))
```
Comment (brc-dd) [inline stories/components/STable.story.vue:155]: Might wanna modify this. I'm not completely sure what's the expected behavior is with filtering multiple items.

File: lib/components/STableCell.vue [hunk 2]
```diff
@@ -5,18 +5,19 @@ import STableCellAvatar from './STableCellAvatar.vue'
...
   className?: string
-  cell?: TableCell | ((value: any, record: any) => TableCell)
+  cell?: any
   value: any
   record: any
   records: Record<string, any>
 }>()
 
-const cell = computed(() =>
+const computedCell = computed<TableCell | undefined>(() =>
```
Comment (brc-dd) [inline lib/components/STableCell.vue:20]: Had to rename this because Volar was getting confused in the template.

**Comment by brc-dd** [comment]
Also there are a bunch of warning/errors on the console when someone filters something 👀  (they were before this PR too).

**Comment by ryo-gk** [comment]
Thanks a lot 🙌 
Looking good to me!

@kiaking 
Could you check, and tweak some designs if you need?

File: lib/components/STableCell.vue (hunk 2 above)
Comment (kiaking) [reply lib/components/STableCell.vue:20]: Ah, gotcha. No problem 👍

File: stories/components/STable.story.vue (hunk 1 above)
Comment (kiaking) [reply stories/components/STable.story.vue]: Ah, well this stuff depends on the business logic anyway so... Let's come back if we find issues on it 👀

**Comment by kiaking** [comment]
> Also there are a bunch of warning/errors on the console when someone filters something

That's the one coming from Histoire I think... not sure what it is. It doesn't happen on actual app that uses STable component. It would be cool if we can fix it though 🤔

**Pull request merged**